parlai/utils/world_logging.py

```diff
@@ -81,15 +81,21 @@ def _add_episode(self, episode):
         self._logs.append(episode)
 
     def _is_batch_world(self, world):
-        return isinstance(world, BatchWorld) and len(world.worlds) > 1
+        return (
+            isinstance(world, BatchWorld) or isinstance(world, DynamicBatchWorld)
+        ) and len(world.worlds) > 1
 
     def _log_batch(self, world):
         batch_act = world.get_acts()
         parleys = zip(*batch_act)
         for i, parley in enumerate(parleys):
-            self._add_msgs(parley, idx=i)
+            # in dynamic batching, we only return `batchsize` acts, but the
+            # 'dyn_batch_idx' key in the task act corresponds the episode index
+            # in the buffer
+            idx = parley[0]['dyn_batch_idx'] if 'dyn_batch_idx' in parley[0] else i
+            self._add_msgs(parley, idx=idx)
             if world.worlds[i].episode_done():
```

Comment:
does this need to be `world.worlds[idx]` as well?

Reply:
yes great catch thank you